hookman/hooks.py

```diff
@@ -0,0 +1,142 @@
+import ctypes
+import inspect
+from pathlib import Path
+from typing import Callable, List
+
+import strictyaml
+
+
+class HooksSpecs():
+    """
+    A class that holds the specification of the hooks, currently the following specification are available:
+    - Project Name:
+        This field will be used to identify the project and to name the hook functions
+    - Version:
+        The current version of the spec, when a new hook is created or modified this version should be incremented
+    - Hooks:
+        A list with the hooks available for the project, each hook is represented by a python function
+    """
+
+    def __init__(self, *, project_name: str, version: str, pyd_name: str, hooks: List[Callable]) -> None:
+        [self._check_hook_arguments(hook) for hook in hooks]

```

Comment:
It is considered bad style to use a list comprehension just for the implicit loop, and discard the results. Just loop over it instead:

```python
for hook in hooks:
    self._check_hook_arguments(hook) 
```

Reply:
- [x] Applied